server/routes/visitJourney/dateAndTime.ts

```diff
@@ -24,65 +24,54 @@ export default class DateAndTime {
     })
 
     let restrictionChangeMessage = ''
-    let matchingSlot = ''
-    let selectedSlot
+    let matchingSlot
 
+    // first time here on update journey, visitSlot.id will be ''
     if (isUpdate && visitSessionData.visitSlot?.id === '') {
-      selectedSlot = getSelectedSlotByStartTimestamp(
+      matchingSlot = getSlotByStartTimeAndRestriction(
         slotsList,
         visitSessionData.visitSlot.startTimestamp,
         visitSessionData.visitRestriction,
       )
 
-      if (selectedSlot) {
-        matchingSlot = selectedSlot.id
+      if (
+        matchingSlot &&
+        (matchingSlot.availableTables > 0 ||
+          visitSessionData.visitRestriction === visitSessionData.originalVisitSlot.visitRestriction)
+      ) {
+        visitSessionData.visitSlot.id = matchingSlot.id
       }
 
-      if (visitSessionData.visitRestriction !== visitSessionData.originalVisitSlot?.visitRestriction) {
-        if (!selectedSlot || selectedSlot.availableTables === 0) {
-          if (visitSessionData.visitRestriction === 'CLOSED') {
-            if (visitSessionData.closedVisitReason === 'visitor') {
-              restrictionChangeMessage =
-                'A new visit time must be selected as this is now a closed visit due to a visitor restriction.'
-            } else {
-              restrictionChangeMessage =
-                'A new visit time must be selected as this is now a closed visit due to a prisoner restriction.'
-            }
-          } else {
-            restrictionChangeMessage = 'A new visit time must be selected as this is now an open visit.'
-          }
-        } else if (selectedSlot && selectedSlot.availableTables > 0) {
-          if (visitSessionData.visitRestriction === 'CLOSED') {
-            if (visitSessionData.closedVisitReason === 'visitor') {
-              restrictionChangeMessage =
-                'This is now a closed visit due to a visitor restriction. The visit time can stay the same.'
-            } else {
-              restrictionChangeMessage =
-                'This is now a closed visit due to a prisoner restriction. The visit time can stay the same.'
-            }
-          } else {
-            restrictionChangeMessage = 'This is now an open visit. The visit time can stay the same.'
-          }
+      if (visitSessionData.visitRestriction !== visitSessionData.originalVisitSlot.visitRestriction) {
```

Comment:
(Hopefully!) just simplifying the existing logic here.

Reply:
why is this not optional ? any more

Reply:
I figured it was unnecessary as although `originalVisitSlot` is [defined as optional](https://github.com/ministryofjustice/book-a-prison-visit-staff-ui/blob/af7e5a5e5eec1fbef8a3805113ae28259d1d080e/server/%40types/bapv.d.ts#L187) for `visitSessionData`, when you start an update journey it always gets set - see [visit.ts](https://github.com/ministryofjustice/book-a-prison-visit-staff-ui/blob/af7e5a5e5eec1fbef8a3805113ae28259d1d080e/server/routes/visit.ts#L113).